Empty State Component for Search

filip-solecki opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

We want to make empty states feel nicer by using a subtle hint that the page will eventually contain data by using a faded row pattern, while also creating space for the middle of the page to show an informative or educational modal.

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/43747
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Log in to the app
2. Go to `Search` page from `LHP`
3. Go to `Workspaces`
4. Enable `Tags` and open `Tags` page
5. Open `Categories` page and remove all categories 

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
6. Upload an image via copy paste
7. Verify a modal appears displaying a preview of that image
--->

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->
<img width="432" alt="Screenshot 2024-06-25 at 12 02 00" src="https://github.com/Expensify/App/assets/153545991/e575b60d-3e31-467c-84df-8ced826b598d">

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->
<img width="431" alt="Screenshot 2024-06-25 at 12 03 54" src="https://github.com/Expensify/App/assets/153545991/955f8248-2c4f-466d-829a-a4b1314f0d4e">

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->
![Simulator Screenshot - iPhone 15 Plus - 2024-06-25 at 12 10 25](https://github.com/Expensify/App/assets/153545991/ff5f3d91-ca98-4732-91a7-1b9c26e0b840)

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->
![Simulator Screenshot - iPhone 15 Plus - 2024-06-25 at 12 12 06](https://github.com/Expensify/App/assets/153545991/6be0c375-c31d-42e3-8a83-4f948b4392ca)

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->
![Screenshot 2024-06-25 at 12 12 22](https://github.com/Expensify/App/assets/153545991/38c206d8-33a3-4d54-b42e-b1256aa74d4e)

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->
![Screenshot 2024-06-25 at 12 14 12](https://github.com/Expensify/App/assets/153545991/0426063c-6ef4-492c-a16c-44329d7495aa)

</details>

filip-solecki committed:
Empty State Component for Search

filip-solecki committed:
Remove backgroundColor

filip-solecki committed:
Animation case

filip-solecki committed:
Add animation and implement on Search page

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki committed:
Small fixes on search empty page, add new empty component on tags and category pages

filip-solecki committed:
Small fixes

filip-solecki committed:
remove unnecessary expensicon

filip-solecki committed:
Adjust to final design

filip-solecki committed:
Fix video styling

filip-solecki committed:
remove unnecessary styles

filip-solecki committed:
self review fixes

filip-solecki committed:
Add height for illustrations

filip-solecki committed:
Fix lint

filip-solecki committed:
Use RN StyleSheet flatten function

kosmydel commented on src/styles/index.ts:
```diff
@@ -5032,6 +5033,44 @@ const styles = (theme: ThemeColors) =>
+        }),
+
+        emptyStateContent: (isSmallScreenWidth: boolean) => ({
+            width: isSmallScreenWidth ? '100%' : 400,
```
Shouldn't `400` be a variable? Is it possible, that it will be used in different places?

filip-solecki replied:
I don't think it will be reused somewhere

kosmydel commented on src/styles/index.ts:
```diff
@@ -1668,6 +1668,7 @@ const styles = (theme: ThemeColors) =>
 
         welcomeVideoNarrowLayout: {
             width: variables.onboardingModalWidth,
+            height: 500,
```
Shouldn't "height" also be a variable if "width" is?

kosmydel reviewed (approved):
LGTM, I've left some minor comments 🚀

filip-solecki commented on src/components/EmptyStateComponent/index.tsx:
```diff
@@ -0,0 +1,102 @@
+import CONST from '@src/CONST';
+import type {EmptyStateComponentProps, VideoLoadedEventType} from './types';
+
+const VIDEO_ASPECT_RATIO = 400 / 225;
```
It may be different for each usage. Here it is adjusted to the "Modal" size

mountiny replied:
It would be great to leave a comment about this, how did we get to this ratio?

filip-solecki committed:
cr fixes

filip-solecki commented on src/pages/Search/EmptySearchView.tsx:
```diff
@@ -1,14 +1,21 @@
+            headerMediaType="illustration"
+            headerMedia={Illustrations.EmptyState}
+            headerStyles={styles.emptyFolderBG}
+            headerContentStyles={{width: 184, height: 112}}
```
Sure, I've added it to styles. Don't think it has to be const as it is strictly connected to this icon

filip-solecki commented on src/components/EmptyStateComponent/index.tsx:
```diff
@@ -0,0 +1,102 @@
+        default:
+            HeaderComponent = null;
+            break;
+    }
```
Added

shawnborton commented:
Can you vertically and horizontally center the illustration here? 
![CleanShot 2024-06-25 at 16 40 56@2x](https://github.com/Expensify/App/assets/2319350/fa1b13fd-3110-4303-a2f8-5cccfb0a5aa9)

shawnborton commented:
How much padding are you using on this part of the component? Feels like there is a bit too much padding at the bottom:
![CleanShot 2024-06-25 at 16 45 08@2x](https://github.com/Expensify/App/assets/2319350/c84c0a59-4564-46ff-b083-871558051e81)

shawnborton commented:
Can we check the padding of the search skeleton rows in the BG? Seems like there is too much left padding there compared to vertical padding above/below the receipt thumbnail:
![CleanShot 2024-06-25 at 16 46 17@2x](https://github.com/Expensify/App/assets/2319350/2d64f0fa-21c7-496e-8259-55bc2d5d0f38)

shawnborton commented:
It looks like the right side of the search skeleton rows has a harsh cut off too:
![CleanShot 2024-06-25 at 17 14 25@2x](https://github.com/Expensify/App/assets/2319350/084c5622-b0aa-4a56-a098-8400eaf7001d)

shawnborton commented:
These receipt thumbnails (on desktop) should be 36x40 - they also seem off:

![CleanShot 2024-06-25 at 17 15 17@2x](https://github.com/Expensify/App/assets/2319350/c1a12cff-9b48-4a99-927f-568f39a89f43)

filip-solecki commented:
> Can you vertically and horizontally center the illustration here?
It is now, I didn't see this doing screenshot

filip-solecki committed:
Remove redundant margin

filip-solecki commented:
> How much padding are you using on this part of the component? Feels like there is a bit too much padding at the bottom:

Yes, it was too big, I've added 20px margin to subtitle and the whole content has 32px padding so it was 52 at the bottom. I've removed 20px margin and it is 32px right now

filip-solecki commented:
Do you want me to adjust the background skeleton? I didn't touch anything there so it is as it was before 😄 
But ofc I can fix it, should I use [this skeleton](https://www.figma.com/design/qQIIaxVUdtggrh8eg8bI5B/Brand-Guidelines-%26-Product-UI?node-id=9112-33415&t=TT5Fni8JR49P0QJr-4)?

shawnborton commented:
Yeah, if you don't mind fixing it, that would be great. You can use what you have linked from Figma but I wouldn't include the left-most checkbox skeleton right now, since we don't have checkboxes on the Search page just yet.

eVoloshchak commented on src/components/EmptyStateComponent/index.tsx:
```diff
@@ -0,0 +1,99 @@
+    );
+}
+
+export default EmptyStateComponent;
```
```suggestion
EmptyStateComponent.displayName = 'EmptyStateComponent';
export default EmptyStateComponent;
```

eVoloshchak commented on src/components/EmptyStateComponent/index.tsx:
```diff
@@ -0,0 +1,99 @@
+
+    const HeaderComponent = useMemo(() => {
+        switch (headerMediaType) {
+            case 'video':
```
We can move 'video',  'illustration',  'animation' strings to `CONST`, then use them here and in `src/components/EmptyStateComponent/types.ts`

filip-solecki replied:
Done

eVoloshchak commented on src/components/EmptyStateComponent/index.tsx:
```diff
@@ -0,0 +1,99 @@
+                    <View style={styles.p8}>
+                        <Text style={[styles.textAlignCenter, styles.textHeadlineH1, styles.mb2]}>{title}</Text>
+                        <Text style={[styles.textAlignCenter, styles.textSupporting, styles.textNormal]}>{subtitle}</Text>
+                        {buttonText && buttonAction && (
```
```suggestion
                        {!!buttonText && !!buttonAction && (
```

eVoloshchak commented on src/components/OptionsListSkeletonView.tsx:
```diff
@@ -17,12 +17,14 @@ function getLinedWidth(index: number): string {
 
 type OptionsListSkeletonViewProps = {
     shouldAnimate?: boolean;
+    gradientOpacity?: boolean;
```
The name is a bit confusing, it suggests this prop should be a number
Would `gradientOpacityEnabled` be better?

filip-solecki replied:
Sounds good, renamed

eVoloshchak commented on src/components/EmptyStateComponent/index.tsx:
```diff
@@ -0,0 +1,99 @@
+function EmptyStateComponent({SkeletonComponent, headerMediaType, headerMedia, buttonText, buttonAction, title, subtitle, headerStyles, headerContentStyles}: EmptyStateComponentProps) {
+    const styles = useThemeStyles();
+    const {isSmallScreenWidth} = useWindowDimensions();
+
```
```suggestion
```

eVoloshchak commented on src/components/Skeletons/ItemListSkeletonView.tsx:
```diff
@@ -10,22 +10,65 @@ type ListItemSkeletonProps = {
+    }
+
+    const flattenStyle = StyleSheet.flatten(style);
+
```
```suggestion
```

eVoloshchak commented on src/components/Skeletons/ItemListSkeletonView.tsx:
```diff
@@ -10,22 +10,65 @@ type ListItemSkeletonProps = {
+            const totalHeight = event.nativeEvent.layout.height;
+
+            const newNumItems = Math.ceil(totalHeight / totalItemHeight);
+
```
```suggestion
```

eVoloshchak commented on src/components/Skeletons/ItemListSkeletonView.tsx:
```diff
@@ -10,22 +10,65 @@ type ListItemSkeletonProps = {
+            }
+
+            const totalHeight = event.nativeEvent.layout.height;
+
```
```suggestion
```

eVoloshchak commented on src/styles/index.ts:
```diff
@@ -5032,6 +5032,50 @@ const styles = (theme: ThemeColors) =>
+            justifyContent: 'center',
+            alignItems: 'center',
+            height: '100%',
+            padding: isSmallScreenWidth ? 24 : 0,
```
Why 24? Is this `variables.componentBorderRadiusRounded`?

filip-solecki replied:
It is value taken from screenshot 😄 
I tried to reproduce screenshots attached to the issue and 24px looks very similar

filip-solecki committed:
Add displayName

Co-authored-by: Eugene Voloshchak <copyreading@gmail.com>

filip-solecki committed:
Add double negation for checks

Co-authored-by: Eugene Voloshchak <copyreading@gmail.com>

filip-solecki committed:
remove empty line

Co-authored-by: Eugene Voloshchak <copyreading@gmail.com>

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki committed:
rename to gradientOpacityEnabled

filip-solecki committed:
Remove redundant empty lines

filip-solecki committed:
Move headerMediaType values to CONST

filip-solecki committed:
Fix lint

filip-solecki committed:
Adjust Search desktop skeleton

filip-solecki commented:
@eVoloshchak All your comments are addressed!

filip-solecki commented:
kind bump @eVoloshchak

eVoloshchak reviewed (commented):
`You haven't created any categories` component is shown while categories are loading, this doesn't look right

1. Sign out, sight back in 
2. Go to Settings -> Workspaces -> select a workspace that has some categories
3. Go to Categories
4. Notice `You haven't created any categories` modal is shown while categories are loading



https://github.com/Expensify/App/assets/9059945/f612e4f5-a593-4546-b037-38ef0be2ac7e

filip-solecki commented:
It is not caused by this PR(reproducible also on staging with old empty state component), although I will take a look

eVoloshchak commented:
- This page should have a scroll view
- Not sure if the modal should shrink in any way if you resize the window, but it shouldn't be on top of `Get a better overview of where money is being spent. Use our default categories or add your own.` text

https://github.com/Expensify/App/assets/9059945/7891d9f1-a73b-4206-ae6c-d5fed02cbe0a

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki commented:
> `You haven't created any categories` component is shown while categories are loading, this doesn't look right
> 
> 1. Sign out, sight back in
> 2. Go to Settings -> Workspaces -> select a workspace that has some categories
> 3. Go to Categories
> 4. Notice `You haven't created any categories` modal is shown while categories are loading

I've found the root cause - it is [here](https://github.com/Expensify/App/pull/41344/files#diff-c42ce6620ce9f24877cffda579d486f8c60ebb0ad577eb5dfcdcb9584208c915:~:text=const%20isLoading%20%3D%20!isOffline%20%26%26%20policyCategories%20%3D%3D%3D%20null%3B). Currently isLoading state is not working properly. Don't know why author changed it but I've asked there

shawnborton commented:
> Not sure if the modal should shrink in any way if you resize the window, but it shouldn't be on top of Get a better overview of where money is being spent. Use our default categories or add your own. text

This is a fair point. Can we give the empty state a min and max height to prevent some of these edge cases?

filip-solecki committed:
Fix overflowing modal

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki commented:
https://github.com/Expensify/App/assets/153545991/92d7bff6-9aed-4dab-b514-3a259d705174

WDYT?
@shawnborton @eVoloshchak

shawnborton commented:
Lovely, I think that's a good improvement.

trjExpensify commented:
Excellent! Is this ready for another review by @eVoloshchak now?

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki commented:
It is!

eVoloshchak commented on src/components/Search/index.tsx:
```diff
@@ -99,7 +99,7 @@ function Search({query, policyIDs, sortBy, sortOrder}: SearchProps) {
                     hash={hash}
                 />
-                <TableListItemSkeleton shouldAnimate />
+                <SearchRowSkeleton shouldAnimate />{' '}
```
What is the purpose of `{' '}` here?

eVoloshchak commented on src/components/Skeletons/SearchRowSkeleton.tsx:
```diff
@@ -96,56 +97,86 @@ function TableListItemSkeleton({shouldAnimate = true, fixedNumItems}: TableListI
+                        // 24 is the padding of the central pane summing two sides
+                        // 12 is the gap between the element and the right button
+                        // 80 is the width of the element itself
+                        x={windowWidth - 375 - 80 - 12 - 24 - 12 - 80}
```
Let's define all of those numbers as constants since we're using them more than once

filip-solecki replied:
Done!

shawnborton commented:
Let's keep the momentum up here, we're so close!

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki committed:
Remove unnecessary empty string

filip-solecki committed:
Move calcuations values to consts

filip-solecki commented:
@eVoloshchak all fixed

eVoloshchak commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

shawnborton commented:
@filip-solecki this screenshot here looks incorrect:
![CleanShot 2024-07-09 at 16 27 49@2x](https://github.com/Expensify/App/assets/2319350/d498235c-4edb-434d-87c0-5b14c35addc7)

We should be maintaining the max-width of the middle modal portion. Can you take a look at that please? Thanks!

filip-solecki committed:
Merge branch 'main' into filip-solecki/empty-state-component

filip-solecki committed:
Fix styling

filip-solecki commented:
@shawnborton WDYT?
![Screenshot 2024-07-10 at 08 13 51](https://github.com/Expensify/App/assets/153545991/49e08ea9-f2cf-466c-bd67-3fa37aeb39ae)

shawnborton commented:
Much better - we should always be using the same exact max-width no matter the platform. 

It does look like the padding around the empty state rows is off here though... as in, why don't they line up with the left edge of the buttons above?
![CleanShot 2024-07-10 at 08 33 34@2x](https://github.com/Expensify/App/assets/2319350/54d9983b-7bf5-4153-b934-ae64a08188f5)

filip-solecki committed:
Fix skeleton styling

filip-solecki commented:
Fixed
![Screenshot 2024-07-10 at 09 22 39](https://github.com/Expensify/App/assets/153545991/48746fbf-a404-4787-aa69-91e63682f23e)

shawnborton commented:
Cool, looking better. In the categories example though, why don't we have the default text above the table like we do when there are categories available? 
![CleanShot 2024-07-10 at 09 27 55@2x](https://github.com/Expensify/App/assets/2319350/889fbca2-b2c5-49ff-88ee-53a0161cc1c3)

I wouldn't think we'd get rid of that text - we're just replacing the table portion here.

filip-solecki commented:
I don't know why the text is missing from my screenshot - I didn't remove it and it's still there, even if the user doesn't have a category to view 😄

shawnborton commented:
Hmm interesting, so are you saying the text is there or it isn't there? I will spin up a test build though and take a look too.

filip-solecki commented:
It is
![Screenshot 2024-07-10 at 09 37 38](https://github.com/Expensify/App/assets/153545991/27f7e664-8521-4bd0-9298-8bf19f546920)

shawnborton commented:
Ok yeah that looks much better! Weird that it was missing originally.

filip-solecki commented:
Ok, there is 24px right now, I'll update in a sec

filip-solecki committed:
Fix padding for empty state modal

filip-solecki commented:
![Screenshot 2024-07-10 at 10 28 20](https://github.com/Expensify/App/assets/153545991/2a96baa6-4294-41d1-8f23-36f600be1490)

shawnborton commented:
Lovely! Okay, l think we should get this thing into final review then?

filip-solecki commented:
Awesome, I see @eVoloshchak didn't finish the checklist

eVoloshchak reviewed (approved):
LGTM!

mountiny commented on src/components/Skeletons/SearchRowSkeleton.tsx:
```diff
@@ -5,23 +5,39 @@ import useWindowDimensions from '@hooks/useWindowDimensions';
 
-function TableListItemSkeleton({shouldAnimate = true, fixedNumItems}: TableListItemSkeletonProps) {
+// 375 is the width of the left pane
+const leftPaneWidth = 375;
```
There must be a const or something for this

filip-solecki replied:
I set video ratio to match modal with animation size. There is no example video or something I can relate to

mountiny reviewed (dismissed):
Thank you! Small comments

filip-solecki committed:
Use variables for LHP width calculations

mountiny reviewed (approved):
Thank you

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

shawnborton commented:
So this is looking awesome @filip-solecki but I found one little bug here - it seems like the layout slightly shifts when the empty state loads right after the loading skeleton:

https://github.com/user-attachments/assets/26627c1d-b00f-4366-81be-9c1a1f851c85

Seems like we might just need some more horizontal padding around the loading skeleton? The whole view should be wrapped with 20px horizontal padding, but it looks like there is a difference between the two. Do you mind creating a follow up PR when you get a moment? Thanks!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9946176278) to staging by https://github.com/mountiny in version: 9.0.7-3 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Cherry-picked](https://github.com/Expensify/App/actions/runs/9947036855) to staging by https://github.com/thienlnam in version: 9.0.7-4 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

@Expensify/applauseleads please QA this PR and check it off on the [deploy checklist](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3AStagingDeployCash) if it passes.

shawnborton commented:
Also looks like we missed a spot in the workspace editor for this new style too:
![CleanShot 2024-07-16 at 07 29 13@2x](https://github.com/user-attachments/assets/ce526c0f-fd29-4358-9d6c-569bdd88ccbf)

filip-solecki commented:
Sure, I'll prepare follow-up PR probably today

shawnborton commented:
Perfect, thanks Filip!

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9965822856) to production by https://github.com/thienlnam in version: 9.0.7-8 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9992557463) to production by https://github.com/mountiny in version: 9.0.8-6 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

shawnborton commented:
Friendly bump on those follow up items @filip-solecki

filip-solecki commented:
Only[ this comment](https://github.com/Expensify/App/pull/44137#issuecomment-2229350014) left right?

shawnborton commented:
I think so. We implemented this for report fields, right?

filip-solecki commented:
Yes, [here](https://github.com/Expensify/App/pull/46476)

shawnborton commented:
Cool, thanks for confirming!
